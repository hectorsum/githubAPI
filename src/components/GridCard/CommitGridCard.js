import React from 'react'
import { useFetchCommits } from '../../hooks/useFetchCommits'
import { getPetTime } from '../../helpers/getPetTime';
import styled from 'styled-components';
import { CommitCard } from '../Card/CommitCard';

export const CommitGridCard = () => {
  const {commits,loading} = useFetchCommits();
  if (loading){
    console.log('loading');
  }else{
    console.log(commits);
    // console.log(getPetTime(commits));
  }
  return (
    <CommitGridCardSection className="container mt-4 ">
      <Title>Github API - Hector Herrera 👨‍💻</Title>
      <Underline/>
      {
        loading && <h1 className="animate__animated animate__flash">Loading..</h1>
      }
      <div className="col-lg-12">
        <GridCard className="row">
          {
            commits.map(commit => {
              return <CommitCard key={commit.sha} {...commit}/>
            })
          }
        </GridCard>
      </div>
    </CommitGridCardSection>
  )
}
const Title = styled.h1 `
  color: ${props => props.theme.titleColor}
`;

const Underline = styled.hr `
  color:white;
`;

const GridCard = styled.div `

`;

const CommitGridCardSection = styled.div`
  
`;