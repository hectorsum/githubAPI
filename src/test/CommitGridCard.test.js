import React from "react";
import "@testing-library/jest-dom";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { CommitGridCard } from "../components/GridCard/CommitGridCard";
import { createSerializer } from "enzyme-to-json";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Testing <CommitGridCard/>", () => {
  const wrapper = shallow(<CommitGridCard />);
  test("Should show <CommitGridCard/> component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
