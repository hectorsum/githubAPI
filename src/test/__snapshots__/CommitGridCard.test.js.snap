// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CommitGridCard/> Should show <CommitGridCard/> component correctly 1`] = `
<styled.div
  className="container mt-4 "
>
  <styled.h1>
    Github API - Hector Herrera 👨‍💻
  </styled.h1>
  <styled.hr />
  <styled.div
    className="animate__animated animate__flash"
  >
    Loading..
  </styled.div>
  <div
    className="col-lg-12"
  >
    <styled.div
      className="row animate__animated animate__fadeIn"
    />
  </div>
</styled.div>
`;
